src/core/Expander/Expander/Expander.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { Component, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import { AutoId } from '../../../utils/AutoId';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { HtmlDiv } from '../../../reset';
+import { baseStyles } from './Expander.baseStyles';
+import {
+  ExpanderGroupConsumer,
+  ExpanderGroupProviderState,
+} from '../ExpanderGroup/ExpanderGroup';
+
+const baseClassName = 'fi-expander';
+const openClassName = `${baseClassName}--open`;
+
+export interface ExpanderProviderState {
+  /** Callback for communicating ExpanderTitle button event to Expander  */
+  onToggleExpander: () => void;
+  /** Open state for expander */
+  open: boolean;
+  /** Id for expander button */
+  titleId: string | undefined;
+  /** Id for expander content */
+  contentId: string | undefined;
+}
+
+const defaultProviderValue: ExpanderProviderState = {
+  onToggleExpander: () => null,
+  open: false,
+  titleId: undefined,
+  contentId: undefined,
+};
+
+const {
+  Provider: ExpanderProvider,
+  Consumer: ExpanderConsumer,
+} = React.createContext(defaultProviderValue);
+
+interface InternalExpanderProps {
+  /**
+   * Children, extend type ExpanderTitleBaseProps or ExpanderContentBaseProps
+   * ExpanderProviderState context is used to communicate between title, content and expander
+   */
+  children: ReactNode;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /**
+   * Id for expander, must be unique. Duplicate id's break ExpanderGroup functionality.
+   * Autogenerated if not provided
+   */
+  id?: string;
+  /** Default status of expander open
+   * @default false
+   */
+  defaultOpen?: boolean;
+  /** Controlled open property */
+  open?: boolean;
+  /** Event handler to execute when clicked */
+  onOpenChange?: (open: boolean) => void;
+  consumer?: ExpanderGroupProviderState;
+}
+
+export interface ExpanderProps extends InternalExpanderProps, TokensProp {}
+
+export interface ExpanderTitleBaseProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Expander consumer for open state and toggle open callback */
+  consumer: ExpanderProviderState;
+}
+
+export interface ExpanderContentBaseProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Expander consumer for open state */
+  consumer: ExpanderProviderState;
+}
```

Comment:
And this one to be in
`suomifi-ui-components/src/core/Expander/ExpanderContent/ExpanderContent.tsx`?

Reply:
See above comment reply.